sw/device/tests/BUILD

```diff
@@ -0,0 +1,479 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("//rules:opentitan.bzl", "OPENTITAN_CPU", "opentitan_binary", "opentitan_functest")
+
+opentitan_functest(
+    name = "aes_smoketest",
+    srcs = ["aes_smoketest.c"],
+    deps = [
+        ":aes_smoketest_lib",
+        "//sw/device/lib/base:mmio",
+    ],
+)
+
+cc_library(
+    name = "aes_smoketest_lib",
+    target_compatible_with = [OPENTITAN_CPU],
+    deps = [
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/exts/common",
+        "//sw/device/lib:irq",
+        "//sw/device/lib:pinmux",
+        "//sw/device/lib/arch:device",
+        "//sw/device/lib/crt",
+        "//sw/device/lib/dif:aes",
+        "//sw/device/lib/dif:gpio",
+        "//sw/device/lib/dif:spi_device",
+        "//sw/device/lib/dif:uart",
+        "//sw/device/lib/runtime:hart",
+        "//sw/device/lib/runtime:log",
+        "//sw/device/lib/runtime:print",
+        "//sw/device/lib/testing:entropy_testutils",
+        "//sw/device/lib/testing/test_framework",
+        "//sw/device/lib/testing/test_framework:ottf",
+    ],
+)
```

Comment:
Is there a reason to structure these as separate targets?  This 'lib' doesn't have any srcs of its own, so its  just a list of deps for the test program and quite a few of these deps are automatically part of the `opentitan_functest` rule.

Reply:
I didn't know that about the `opentitan_functest` rule, and hadn't yet worked out the organizing principle for when we do or don't split things. You're right that combining them simplifies this PR, and there's no reason for splitting them.